tests: nffs: Tweak Kconfig to fit in RAM

carlescufi opened the pull request:
NFFS tests do not fit in RAM on nRF5x, tweak the configurations so that
they do.

Fixes: #12065

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
tests: nffs: Tweak Kconfig to fit in RAM

NFFS tests do not fit in RAM on nRF5x, tweak the configurations so that
they do.

Fixes: #12065

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

galak reviewed (changes requested):
Can you fix this so it runs on sanitycheck for the nrf boards.  testcase.yaml needs updating.

carlescufi committed:
tests: nffs: Run NFFS tests on nRF52840

After the builds have been fixed, enable the tests on the 256KB-RAM
Nordic IC.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@galak done

galak commented on tests/subsys/fs/nffs_fs_api/basic/testcase.yaml:
```diff
@@ -3,4 +3,4 @@ common:
   filesystem.nffs.basic:
     extra_args: TEST=basic
-    platform_whitelist: qemu_x86
+    platform_whitelist: qemu_x86 nrf52840_pca10056
```
Should this also include: nrf51_pca10028 & nrf52_pca10040?

Looking at tests/subsys/fs/nffs_fs_api/basic/CMakeLists.txt I see:

```
if(BOARD STREQUAL nrf51_pca10028)
    set(CONF_FILE nrf5x.conf)
elseif(BOARD STREQUAL nrf52_pca10040)
    set(CONF_FILE nrf5x.conf)
elseif(BOARD STREQUAL nrf52840_pca10056)
    set(CONF_FILE nrf5x.conf)
endif()
```

carlescufi replied:
They won't fit in RAM, the test requires 230KB of RAM and the 52840 is the only one with 256KB

galak replied:
> They won't fit in RAM, the test requires 230KB of RAM and the 52840 is the only one with 256KB

So should those be removed from the CMakeLists?

carlescufi replied:
I have now

carlescufi committed:
test: nffs: Remove unsupported ICs from CMake files

The smaller Nordic ICs are not supported for those tests due to not
enough RAM.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
recheck

carlescufi commented:
@galak you OK with this now?
